### PR TITLE
Add mixer role and service account

### PR DESCRIPTION
**Pull request by mandarjog**
Add mixer role and service account.

@jmuk can you verify that the permissions needed by the discovery API is covered in the permissions? 
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: install/kubernetes/istio-rbac-beta.yaml [hunk 1]
```diff
@@ -69,6 +87,20 @@ roleRef:
   name: istio-pilot
   apiGroup: rbac.authorization.k8s.io
 ---
+# Grant permissions to Mixer.
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: istio-mixer-admin-role-binding
```
Comment (jmuk) [inline install/kubernetes/istio-rbac-beta.yaml]: There's already a binding of `istio-mixer-service-account` to `istio-pilot` role.
Maybe you don't have to edit this file?

File: install/kubernetes/istio-rbac-beta.yaml [hunk 2]
```diff
@@ -27,6 +27,24 @@ rules:
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: istio-mixer
+rules:
+- apiGroups: ["config.istio.io"]
+  resources: ["*"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list", "watch"]
```
Comment (jmuk) [inline install/kubernetes/istio-rbac-beta.yaml]: Not necessary right now, but for the upcoming external admission hook registrations:
```yaml
apiGroups: ["admissionregistration.k8s.io"]
resources: ["externaladmissionhookconfigurations"]
verbs: ["get", "delete", "create"]
```
will be needed.

Comment (mandarjog) [reply install/kubernetes/istio-rbac-beta.yaml]: We should create a new role for it, mixer itself does not need to do this.

File: install/kubernetes/istio-rbac-beta.yaml (hunk 1 above)
Comment (mandarjog) [reply install/kubernetes/istio-rbac-beta.yaml]: I removed that binding. every component gets own role / service account now.

File: install/kubernetes/istio-rbac-alpha.yaml [hunk 3]
```diff
@@ -24,6 +27,24 @@ rules:
...
+  name: istio-mixer
+rules:
+- apiGroups: ["config.istio.io"] # Istio CRD watcher
+  resources: ["*"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["configmaps", "endpoints", "pods", "services", "namespaces", "nodes", "secrets"]
```
Comment (andraxylia) [inline install/kubernetes/istio-rbac-alpha.yaml]: Does mixer need to read all these resources, including secrets? Can you try to reduce the list?

File: install/kubernetes/istio-rbac-beta.yaml [hunk 4]
```diff
@@ -27,6 +27,24 @@ rules:
...
+  name: istio-mixer
+rules:
+- apiGroups: ["config.istio.io"] # Istio CRD watcher
+  resources: ["*"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["configmaps", "endpoints", "pods", "services", "namespaces", "nodes", "secrets"]
```
Comment (andraxylia) [inline install/kubernetes/istio-rbac-beta.yaml]: same as above, please keep only what is truly needed.

File: install/kubernetes/istio-rbac-alpha.yaml (hunk 3 above)
Comment (mandarjog) [reply install/kubernetes/istio-rbac-alpha.yaml]: yes, mixer will need secrets shortly as a way to get private credentials to adapters.

File: install/kubernetes/istio-rbac-beta.yaml (hunk 2 above)
Comment (jmuk) [reply install/kubernetes/istio-rbac-beta.yaml]: I see. Thanks for the explanations.

File: install/kubernetes/istio-rbac-beta.yaml [hunk 5]
```diff
@@ -27,6 +27,24 @@ rules:
...
+  name: istio-mixer
+rules:
+- apiGroups: ["config.istio.io"] # Istio CRD watcher
+  resources: ["*"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["configmaps", "endpoints", "pods", "services", "namespaces", "secrets"]
```
Comment (jmuk) [inline install/kubernetes/istio-rbac-beta.yaml]: regarding the minimizing the role as @andraxylia mentioned; those are not used by `pkg/config/crd`.
They might be used by kubernetes adapter though?

**Comment by ldemailly** [comment]
merged as it passed rbac tests (and got the usual 503 on the other cluster)
refinements in any can go in followup PRs

**Pull request merged**